pkg/cfn/manager/iam.go

```diff
@@ -15,7 +14,7 @@ import (
 
 // makeIAMServiceAccountStackName generates the name of the iamserviceaccount stack identified by its name, isolated by the cluster this StackCollection operates on and 'addon' suffix
 func (c *StackCollection) makeIAMServiceAccountStackName(namespace, name string) string {
-	return fmt.Sprintf("eksctl-%s-addon-iamserviceaccount-%s-%s", c.spec.Metadata.Name, namespace, name)
+	return c.spec.MakeStackName("addon", "iamserviceaccount", namespace, name)
```

Comment:
I think we do something in Kuberentes part as well, right? Reference this name somewhere in a config map I think?

Reply:
Are you sure it uses the stack name? I was fairly sure the serviceaccount object pointed at the role name instead. I'll go through and check the code / test in a cluster to see what it does to make sure it's not broken.

Reply:
Nope, not certain. :D Thanks. :)

Reply:
So it is only the role name and not the stack name that get used in the service account annotation, everything seems to be working fine from my dev testing too

Reply:
Awesome! Thank you for confirming.